src/restconfig/src/main/java/org/geoserver/rest/catalog/LayerGroupController.java

```diff
@@ -229,6 +229,8 @@ protected void postEncodeReference(Object obj, String ref, String prefix,
                 }
                 if ( obj instanceof LayerInfo ) {
                     converter.encodeLink("/layers/" + converter.encode(ref), writer);
+                } else if ( obj instanceof LayerGroupInfo) {
+                    converter.encodeLink("/layergroups/" + converter.encode(ref), writer);
```

Comment:
Good question - I'd forgotten that layer groups behave differently from layers in this case, so that currently is not handled. Will fix.